### ISSUE TITLE
Allow multiple instances of installed NVDA

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#7765]
Steps to reproduce:
1. Find an issue which is not easy to atribute to NVDA
2. Test if that issue occurs in several installed / portable NVDA versions

Expected behavior:
In order to create high qualitative issues, sometimes comprehensive tests have to be conducted. Therefore, it would significantly ease the tests if multiple instances of NVDA could be installed on a PC. This would also be a good work around for creating  different portable versions of NVDA on a storage device for testing purposes without having to download different versions of NVDA to make portable versions of them.

Actual behavior:
NVDA can create multiple portable versions on a storage device. But for that you need to download the relevant NVDA version since you have to create the portable version from the coresponding NVDA version. A simple NVDA user who wants to create an issue would be frustrated after five new NVDA installations or five or six downloads of each version he wants to create a portable version of. So in my case, I would install 5 or 6 versions of NVDA on my system and every time I encounter issues I could test it very fast with earlier versions. Apart from that, I would not have to download every time the version of NVDA of which I want to create the portable copy.

System configuration:
NVDA version:
All

NVDA Installed or portable:
Installed

Other information:

Windows version:
All

Name and version of other software in use when reproducing the issue:
NVDA

Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
Yes, all

**Comment by josephsl** [comment]
Hi, no, people can create portable copy without installing one first (as someone mentioned yesterday, I too am wondering if we need to revise our manuals…). Thanks.

**Comment by Adriani90** [comment]
Well I think there is a very short point on that in the manual which explains your point. But I treated the topic from a testing perspective. I edited the thext accordingly to avoid missunderstandings.

**Comment by Brian1Gaff** [comment]
I have loads of different versions and setups on this machine but you have 
to close one before starting the next. It would be nice if you had a switch 
that said running nvda release  blah blah as portable version.
 I have said here before that I use my famous batch file for this now rather 
than the normal start of nvda which can sometimes fail to run due to this 
elusive bug mentioned in another issue.
 However a portable version on windows 10 is of limited use compared to an 
installed version so is there any way we can shut down one installed copy 
and open another.
 I suspect this will fail due to registry issues, also having looked at 
diagnostics during the problem with update failures when Mick wanted to put 
some dependencies into new folders with the name of the snap or version, 
sometimes  strange bits of one copy are still running when another is 
actually supposed to be running as the file is somehow locked.

 I've seen this happen in other software as well even firefox, so Windows is 
a bit uncertain on this kind of thing as all links to, say a dll do not 
vanish simply when you close a piece of software do they?


Maybe you might be able to do it on a different user profile, I've not 
tried.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by Adriani90** [comment]
How did you manage to install different versions of NVDA?

**Comment by Adriani90** [comment]
When I have tried it, my actual installed version gets replaced by the one which I want to install.

**Comment by leonardder** [comment]
I actually don't think this is a very good idea. Note that you can switch from one NVDA instance to another by simply providing the "-r or --replace" command line parameter.

Furthermore, most issues can be tested with portable versions as well. If you really want to test with an installed version, even though officially unsupported, you can replace an installed copy quite easily in under a minute.

Leaning towards a won't fix, but I'd like to have opinions from at least either @michaeldcurran or @feerrenrut on this,

**Comment by michaelDCurran** [comment]
Closing as wontfix. In many cases a portable copy is enough. For more complex cases, consider using virtual machines.
